prusti-utils/src/config.rs

```diff
@@ -190,6 +191,17 @@ lazy_static::lazy_static! {
             allowed_keys.iter().filter(|key| key.len() > MAX_CONFIG_LEN).collect::<Vec<_>>()
         );
 
+        // 0. Apply Lithium overrides if necessary
+        if let Ok(true) = env::var("PRUSTI_NATIVE_VERIFIER").map(|s| s.parse::<bool>().unwrap_or(false)) {
+            settings.set_default("viper_backend", "Lithium").unwrap();
+            settings.set_default("verify_specifications_backend", "Lithium").unwrap();
+            settings.set_default("encode_bitvectors", true).unwrap();
+            settings.set_default("enable_purification_optimization", true).unwrap();
+            settings.set_default("unsafe_core_proof", true).unwrap();
+            settings.set_default("verify_core_proof", false).unwrap();
+            settings.set_default("inline_caller_for", true).unwrap();
+        }
+
```

Comment:
Where is `PRUSTI_NATIVE_VERIFIER` used?

Reply:
Grouping all required flags into one, to facilitate switching between the backends. Alternatively I could check if `viper_backend == "Lithium"`.

Why an environment variable? For some reason if I do

```rust
if settings.get_bool("native_verifier").unwrap_or(false) {
   ...
}
```

at the end (after the `-P` arguments) and provide it via `cargo-prusti -- -Pnative_verifier=true` it breaks everything, including the building of `prusti-contracts` 🤨

Reply:
There are various reasons why I'd suggest to undo this change and look for another solution:
* The `native_verifier` and `viper_backend` flags are partially overlapping.
* The code checks the environment variable `PRUSTI_NATIVE_VERIFIER`, but does not work if the same flag is set via `-P`. This goes against the current design of the configuration flags.
* Changing the default of some flags depending on the value of some flags looks like the kind of thing that easily leads to surprises and hard-to-debug issues. We were forced to do it in one case (`DEFAULT_PRUSTI_*`) because of the way cargo works, but in this PR I think that we have more freedom to choose a better solution.

If Lithium *requires* a specific set of flags, what about just implementing runtime checks like we do [here](https://github.com/viperproject/prusti-dev/blob/a5c29c994cee03e1ba02c3bc2c2761803571d3f5/prusti-utils/src/config.rs#L602-L605)? Moreover, if a set of tests require a long combination of flags, what about adding one entry [here](https://github.com/viperproject/prusti-dev/blob/a5c29c994cee03e1ba02c3bc2c2761803571d3f5/prusti-tests/tests/compiletest.rs#L162-L219), with a new test folder such as `prusti-tests/tests/verify_lithium`?